packages/ilmomasiina-backend/src/routes/utils.ts

```diff
@@ -0,0 +1,34 @@
+import { StringifyApi } from '@tietokilta/ilmomasiina-models/src/schema/utils';
+
+export function toDate<T extends string | null | undefined>(s: string | T):
+T extends string ? Date : Exclude<T, string> {
+  // @ts-ignore
+  return (typeof s === 'string') ? new Date(s) : s;
+}
```

Comment:
Avoid `@ts-ignore`. (Yes, this kind of type twiddling is generally a pain in the arse)

```suggestion
export function toDate<T>(s: T): Date | Exclude<T, string> {
  return typeof s === 'string' ? new Date(s) : s as Exclude<T, string>;
}
```

Reply:
Resolved in 8d88854